src/contracts/LidoARM.sol

```diff
@@ -0,0 +1,68 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.23;
+
+import {Initializable} from "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+import {SafeCast} from "@openzeppelin/contracts/utils/math/SafeCast.sol";
+
+import {AbstractARM} from "./AbstractARM.sol";
+import {LidoLiquidityManager} from "./LidoLiquidityManager.sol";
+
+/**
+ * @title Lido (stETH) Automated Redemption Manager (ARM)
+ * @dev This implementation supports multiple Liquidity Providers (LPs) with single buy and sell prices.
+ * It also integrates to a LiquidityProviderController contract that caps the amount of assets a liquidity provider
+ * can deposit and caps the ARM's total assets.
+ * A performance fee is also collected on increases in the ARM's total assets.
+ * @author Origin Protocol Inc
+ */
+contract LidoARM is Initializable, AbstractARM, LidoLiquidityManager {
```

Comment:
I think it's a little awkward having a LidoARM that inherits from both an AbstractARM and the LidoLiquidityManager. I think it would be a bit cleaner if just moved the code from the LidoLiquidityManager up to this level. There would be a little less indirection going on.

I can't think of a near or mid term reason we need the STETH liquidity management separate to be used into another contract.

Reply:
I agree. I've collapsed LidoLiquidityManager into LidoARM